### PR TITLE
Exclude irrelevant tags from resolve_tags() for tracing

### PR DESCRIPTION
**Pull request by xq-yin**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/xq-yin/mlflow/pull/12330?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12330/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12330
```

</p>
</details>

### What changes are proposed in this pull request?

Initial issue is that trace logging fails from databricks notebooks because an irrelevant tag sparkDataSourceInfo exceeds limits.

After discussing with @daniellok-db and @B-Step62, it seems that resolve_tags() was introduced originally to render the traces table in UI. And not all tags are needed for the UI.
this is the current list of tags that's passed to frontend for react component to render the table. 
image
among these, mlflow.databricks.cluster.id is not used by UI, mlflow.trace_schema.version and mlflow.artifactLocation are not generated from resolve_tags(). Based on this info, this PR makes the rest of the tags an allowlist, and remove irrelevant tags if it's not in the list.

The original issue should be solved because the sparkDataSourceInfo isn't in the new allowlist

### How is this PR tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests

pytest tests/tracing/processor/test_mlflow_processor.py
tested in notebook
Verified the traces UI, especially the "source" column still renders, and the link can point to the right notebook
image

### Does this PR require documentation update?

- [X] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [X] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

**Review by BenWilson2 — APPROVED** [review]
Good approach! LGTM!

File: mlflow/tracing/constant.py [hunk 1]
```diff
@@ -34,3 +34,16 @@ class SpanAttributeKey:
 # Key for the trace schema version in the trace. This key is also used in
 # Databricks model serving to be careful when modifying it.
 TRACE_SCHEMA_VERSION_KEY = "mlflow.trace_schema.version"
+
+# The list of tags generated from resolve_tags that are required for tracing UI
+TRACE_RESOLVE_TAGS_ALLOWLIST = (
```
Comment (B-Step62) [inline mlflow/tracing/constant.py:39]: Can we use the constants defined in `mlflow/utils/mlflow_tags.py`?

**Review by B-Step62 — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/12330/files#r1637211981 is addressed!
Thank you for working on this fix!!

File: mlflow/tracing/constant.py (hunk 1 above)
Comment (xq-yin) [reply mlflow/tracing/constant.py]: good suggestion. have incorporated the change

**Review by daniellok-db — APPROVED** [review]
LGTM, thanks!

**Pull request merged**